mirror_test.go

```diff
@@ -39,6 +41,74 @@ func TestGetSleepTime(t *testing.T) {
 
 }
 
+type ResponseContainer struct {
+	TagImageName               string
+	TagImageOptions            docker.TagImageOptions
+	PullImageOptions           docker.PullImageOptions
+	PullImageAuthConfiguration docker.AuthConfiguration
+	PushImageOptions           docker.PushImageOptions
+	PushImageAuthConfiguration docker.AuthConfiguration
+	RemoveImageName            string
+}
+
+type TestDockerClient struct {
+	ResponseContainer *ResponseContainer
+}
+
+func (t *TestDockerClient) Info() (*docker.DockerInfo, error) {
+	return &docker.DockerInfo{}, nil
+}
+
+func (t *TestDockerClient) TagImage(name string, opts docker.TagImageOptions) error {
+	t.ResponseContainer.TagImageName = name
+	t.ResponseContainer.TagImageOptions = opts
+	return nil
+}
+
+func (t *TestDockerClient) PullImage(opts docker.PullImageOptions, authConfig docker.AuthConfiguration) error {
+	t.ResponseContainer.PullImageOptions = opts
+	t.ResponseContainer.PullImageAuthConfiguration = authConfig
+	return nil
+}
+
+func (t *TestDockerClient) PushImage(opts docker.PushImageOptions, auth docker.AuthConfiguration) error {
+	t.ResponseContainer.PushImageOptions = opts
+	t.ResponseContainer.PushImageAuthConfiguration = auth
+	return nil
+}
+
+func (t *TestDockerClient) RemoveImage(name string) error {
+	t.ResponseContainer.RemoveImageName = name
+	return nil
+}
+
+func CreateTestDockerClient(responseContainer *ResponseContainer) *TestDockerClient {
+	return &TestDockerClient{ResponseContainer: responseContainer}
+}
+
+func TestPullImage(t *testing.T) {
+	responseContainer := &ResponseContainer{}
+	var client DockerClient
+	client = CreateTestDockerClient(responseContainer)
+	repo := Repository{
+		PrivateRegistry: "private-registry/",
+		Name:            "elasticsearch",
+	}
+
+	m := mirror{
+		dockerClient: &client,
+	}
+	m.setup(repo)
+	m.pullImage("latest")
+
+	got := responseContainer.PullImageOptions.Repository
+	want := "private-registry/elasticsearch"
```

Comment:
Should we also add a test for when there _isn't_ a private registry?

Reply:
Added a second test for this scenario, they both will run in parallel.